### ISSUE TITLE
Massive CPU Usage

### ISSUE THREAD

**Issue by scrungus** [digimezzo/dopamine-windows#672]
## I'm submitting a...

<pre><code>
[ ] Regression 
[ x] Bug report  
[ ] Feature request
</code></pre>

## I'm using these versions

* Windows version: 10
* Dopamine version: 1.5.10

## Reproduction steps

* Step 1 : start a long shuffle playlist

## Observed behavior

Using around 50% of CPU constantly. I have an i5 6600k. Will stutter if any other program uses a lot of CPU time

## Desired behavior

Use much less

## Log file

![image](https://user-images.githubusercontent.com/33693738/36357440-fe0025e2-14f5-11e8-8918-c41252bf9e75.png)

**Comment by Magentaize** [comment]
Some users also report this issue, but I have no idea and cannot repro.

**Comment by AbdelrahmanHafez** [comment]
+1
playlist with 1000 soundtracks on shuffle

**Comment by fhmr** [comment]
Yep I'm also getting this when playing a large playlist.

**Comment by digimezzo** [comment]
To @scrunguss, @AbdelrahmanHafez, @fhmr and anyone else experiencing this issue. We're having a hard time reproducing this issue. So, if you don't mind, we'd like you to do some investigation on your computer. Here is how you can help:

**1. Download Process Explorer** 
It is a stand-alone tool from Microsoft, which doesn't require to be installed on your computer. This is the download link: [https://docs.microsoft.com/en-us/sysinternals/downloads/process-explorer](https://docs.microsoft.com/en-us/sysinternals/downloads/process-explorer)
You will get a zip file, that you'll have to extract. The file that interests us in the zip is "procexp.exe".

**2. Find treads which use a lot of CPU by using Process Explorer**
Perform this step when Dopamine is started, and when it is consuming a lot of CPU.
Right-click procexp.exe and select "Run as administrator". You'll get a window which looks like this:

![image](https://user-images.githubusercontent.com/11945339/46598642-1a28c080-cae5-11e8-85d2-fdbd020fff2c.png)

In that Window, find Dopamine.exe and double click it. You'll get a window like this:

![image](https://user-images.githubusercontent.com/11945339/46598692-4cd2b900-cae5-11e8-9294-38f6fc27770a.png)

Make sure you select the "Threads" tab. Then, **take a screenshot of that screen**
 When the screenshot is taken, find the thread which uses the most CPU and double click it. That will show you a stack trace for that thread, which looks like this:

![image](https://user-images.githubusercontent.com/11945339/46598781-94f1db80-cae5-11e8-8104-257b6211927e.png)

**Take a screenshot of the stack trace**

Add both screenshots to this bug report.

To summarize, we need 2 screenshots:

**Screenshot 1:** the threads window of Process Explorer
**Screenshot 2:** the stack trace window of Process Explorer, for the thread which uses the most CPU.

Thanks!

**Comment by AbdelrahmanHafez** [comment]
I would love to help on this one, but I have switched media players some time ago, commenting this in hope that @scrunguss @fhmr would help.
Best of luck!

**Comment by digimezzo** [comment]
@AbdelrahmanHafez no problem. Just out of curiousity. Which media player are you now using, and which feature has won you over? :) (I'm just trying to find out where I can improve). Thanks!

**Comment by scrungus** [comment]
Just to let you know, I am currently running 2.0 and no longer seem to be
experiencing the issue. Something in the update must've resolved it.

**Comment by AbdelrahmanHafez** [comment]
@digimezzo I started using foobar2000, I posted [a reddit topic](https://www.reddit.com/r/Windows10/comments/8pkouj/whats_a_good_music_player_for_windows/) asking for people's opinions, and what my required features are.

I am in it mainly for the UX, the fact that I can navigate the app with my keyboard only, and minimize to tray without having to touch my mouse. Other perks are the simple, customizable UI. I started switching the UI in everything that I can to match Atom Dark One theme, and how light it is.

**Comment by SCLeoX** [comment]
@digimezzo 
![dopamine_2018-10-08_16-24-19](https://user-images.githubusercontent.com/13428807/46631846-a2738980-cb16-11e8-80c2-5414393817d7.png)
![taskmgr_2018-10-08_16-23-54](https://user-images.githubusercontent.com/13428807/46631901-c8009300-cb16-11e8-8d5d-067a41f490bd.png)
![procexp64_2018-10-08_16-23-30](https://user-images.githubusercontent.com/13428807/46631858-abfcf180-cb16-11e8-959c-46a5f8bbb442.png)
![procexp64_2018-10-08_16-23-40](https://user-images.githubusercontent.com/13428807/46631882-b9b27700-cb16-11e8-971d-5375a77e46db.png)

My processor is i5-6200U
My playlist contains 5302 songs

**Comment by SCLeoX** [comment]
Dopamine will usually use a bunch of CPU when started. Then, the CPU usage will drop to about 5%-20% which is tolerable. After a couple hours, CPU usage will rise again to above 50%. This time, it will not decrease and the music will have a hard time playing. However, restart Dopamine does solve the problem.

**Comment by rikonaka** [comment]
I found that the default `Spectrum style` configuration may be the cause of excessive CPU and GPU usage, which in the `Setting->Playback`.

**Comment by rikonaka** [comment]
The default setting of the program is `Flames`. After I change to `Stripes`, the high CPU and GPU occupancy rate drops below 10%.
![image](https://user-images.githubusercontent.com/13602602/59330445-0ee8fd00-8d24-11e9-82fb-6391cd686c19.png)
Task Manager
![image](https://user-images.githubusercontent.com/13602602/59330482-2a540800-8d24-11e9-8130-219187ce1d59.png)
When I use the `Flames` setting, the CPU and GPU usage is as high as 30%. After changing to `Stripes`, it is much better now.
I hope this feedback can help you.

**Comment by SCLeoX** [comment]
@rikonaka Thanks for sharing. It is a little bit better. But my usage is still as high as 30% (was about 35%)

edit: nope it is back to 60% after about 2 minutes

**Comment by digimezzo** [comment]
@SCLeoX As your system seems to be affected a lot, could you have a look at my post a bit higher, concerning **Process Explorer**? Could you follow that procedure and post some screenshots about what you see in process explorer. That would be really helpful to troubleshoot this issue. Thanks.

**Comment by SCLeoX** [comment]
@digimezzo If you scroll up a little bit, you can see I already did that about 8 month ago.

**Comment by digimezzo** [comment]
Sorry, I didn't realize it was you. Those screenshots didn't help unfortunately. Is was hoping to get insight on other people's processes to try to understand what is happening.

**Comment by digimezzo** [comment]
@SCLeoX Do you also have massive cpu usage when the spectrum analyzer is disabled completely? (When you toggle it off in the settings).

**Comment by SCLeoX** [comment]
@digimezzo Yes, I distinctively remember I tried that too and it did not solve the issue.

Now I am using a new laptop (with much higher specs), the cpu usage issue is gone completely. (I am using the exact same music library and Dopamine binary as I simply copy-paste the entire folder to my new laptop.)

**Comment by Magentaize** [comment]
@SCLeoX Yep, so the problem has gone away by replacing your old laptop now. As I've said a few month ago, i5-6200U is too weak to run a huge music library and buying a new laptop with cpu model which end of H may be a better choice.
